modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -0,0 +1,580 @@
+     * @param startTime Start time in milliseconds.
+     * @param duration Job execution time.
+     * @param timedOut {@code True} if job is timed out.
+     */
+    public void job(IgniteUuid sesId, long queuedTime, long startTime, long duration, boolean timedOut) {
+        doWrite(OperationType.JOB,
+            () -> 24 + 8 + 8 + 8 + 1,
+            buf -> {
+                writeIgniteUuid(buf, sesId);
+                buf.putLong(queuedTime);
+                buf.putLong(startTime);
+                buf.putLong(duration);
+                buf.put(timedOut ? (byte)1 : 0);
+            });
+    }
+
+    /**
+     * @param op Operation type.
+     * @param sizeSupplier Record size supplier.
+     * @param writer Record writer.
+     */
+    private void doWrite(OperationType op, IntSupplier sizeSupplier, Consumer<ByteBuffer> writer) {
+        FileWriter fileWriter = this.fileWriter;
+
+        // Writer stopping.
+        if (fileWriter == null)
+            return;
+
+        int size = sizeSupplier.getAsInt();
+
+        SegmentedRingByteBuffer.WriteSegment seg = reserveBuffer(fileWriter, op, size);
+
+        if (seg == null)
+            return;
+
+        writer.accept(seg.buffer());
+
+        seg.release();
+    }
+
+    /**
+     * Reserves buffer's write segment.
+     *
+     * @param fileWriter File writer.
+     * @param type Operation type.
+     * @param size Record size.
+     * @return Buffer's write segment or {@code null} if not enought space or writer stopping.
+     */
+    private SegmentedRingByteBuffer.WriteSegment reserveBuffer(FileWriter fileWriter, OperationType type, int size) {
+        SegmentedRingByteBuffer.WriteSegment seg = fileWriter.writeSegment(size + /*type*/ 1);
+
+        if (seg == null) {
+            fileWriter.logSmallBufferMessage();
+
+            return null;
+        }
+
+        // Ring buffer closed (writer stopping) or maximum size reached.
+        if (seg.buffer() == null) {
+            seg.release();
+
+            if (!fileWriter.isCancelled())
+                fileWriter.onMaxFileSizeReached();
+
+            return null;
+        }
+
+        ByteBuffer buf = seg.buffer();
+
+        buf.put((byte)type.ordinal());
+
+        return seg;
+    }
+
+    /** @return Performance statistics file. */
+    public static File statisticsFile(GridKernalContext ctx) throws IgniteCheckedException {
+        String igniteWorkDir = U.workDirectory(ctx.config().getWorkDirectory(), ctx.config().getIgniteHome());
+
+        File fileDir = U.resolveWorkDirectory(igniteWorkDir, PERFORMANCE_STAT_DIR, false);
+
+        return new File(fileDir, "node-" + ctx.localNodeId() + ".prf");
+    }
+
+    /** Writes {@link UUID} to buffer. */
+    public static void writeUuid(ByteBuffer buf, UUID uuid) {
+        buf.putLong(uuid.getMostSignificantBits());
+        buf.putLong(uuid.getLeastSignificantBits());
+    }
+
+    /** Reads {@link UUID} from buffer. */
+    public static UUID readUuid(ByteBuffer buf) {
```

Comment:
This method should be moved to the reader of some utility class.

Reply:
Done.